assets/wire/utils.ts

```diff
@@ -258,34 +258,34 @@ function getTextFromNode(node: Node, count: number): { text: string; remainingCo
 export function shortHighlightedtext(html: string, maxLength = 40) {
     const div = document.createElement('div');
     div.innerHTML = html;
-    const highlightSpan = div.querySelector('span.es-highlight');
-    if (!highlightSpan) {
+    const highlightSpans = Array.from(div.querySelectorAll('span.es-highlight'));
+    if (highlightSpans.length === 0) {
         return html;
     }
 
     let extractedText = '';
 
-    // Extract text from previous siblings
-    let currentNode: Node | null = highlightSpan.previousSibling;
-    while (currentNode) {
-        const {text, remainingCount} = getTextFromNode(currentNode, maxLength);
-        extractedText = text + extractedText;
-        maxLength = remainingCount;
-        currentNode = currentNode.previousSibling;
-    }
+    for (const highlightSpan of highlightSpans) {
+        let currentNode: Node | null = highlightSpan.previousSibling;
+        while (currentNode && currentNode.nodeType === Node.TEXT_NODE) {
```

Comment:
when it would need iterating for more than 1 previous node?

Reply:
correct, the `while` loop checking for previous siblings of `highlightSpan` (previous nodes of type `Node.TEXT_NODE`) may not be necessary and can be simplified since I am just thinking in extracting the text before the `highlightSpan`.